### PR TITLE
Migrate patient search screen to Mobius: Strangulate and remove controller

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/639/migrate-patient-search-screen-to-mobius

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/search/PatientSearchUpdate.kt [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
 import org.simple.clinic.mobius.next
 
 class PatientSearchUpdate : Update<PatientSearchModel, PatientSearchEvent, PatientSearchEffect> {
 
   override fun update(model: PatientSearchModel, event: PatientSearchEvent): Next<PatientSearchModel, PatientSearchEffect> {
-    return when(event) {
+    return when (event) {
       is SearchQueryTextChanged -> next(model.queryChanged(event.text))
+      is SearchClicked -> {
+        val validationErrors = validateInput(model.enteredQuery)
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/search/PatientSearchUpdate.kt]: Can we extract this in a private function?

**Comment by Honey14** [comment]
Tests were running fine in the build. Can you check this once?

File: app/src/main/java/org/simple/clinic/search/PatientSearchUpdate.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/search/PatientSearchUpdate.kt]: It was done in a later commit in the same PR.

**Comment by vinaysshenoy** [comment]
> Tests were running fine in the build. Can you check this once?

I looked at the logs and it seems like it wasn't able to fetch a particular dependency. Running again.

**Pull request merged**